### PR TITLE
Added HW19 Solution.

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**
ADDED: Authorization, Inject data, Role model, Logout.
IN PROGRESS: Remove Cookies and change token to Optional<token>.

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/controller/LogoutController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class LogoutController extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        HttpSession session = req.getSession(true);
+        session.removeAttribute("userId");
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internetshop/controller/LogoutController.java:16]: req.getSession().invalidate();

File: src/main/java/mate/academy/internetshop/controller/LogoutController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        for (Cookie cookie : req.getCookies()) {
+            if (cookie.getName().equals("MATE")) {
+                cookie.setMaxAge(0);
+                cookie.setValue("");
+            }
+            resp.addCookie(cookie);
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internetshop/controller/LogoutController.java]: addCookie теж варто робити в блоці `if`

File: src/main/java/mate/academy/internetshop/controller/LogoutController.java (hunk 1 above)
Comment (AlexanderSashchenko) [reply src/main/java/mate/academy/internetshop/controller/LogoutController.java:16]: Fixed.

File: src/main/java/mate/academy/internetshop/controller/LogoutController.java (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/java/mate/academy/internetshop/controller/LogoutController.java]: Deleted cookies at all;

**Review by Tarabarnik — APPROVED** [review]

**Pull request merged**